[NFC] Cleanup array syntax in tests directory

colemanw opened the pull request:
Overview
----------------------------------------
As long as the tests still pass, this is simpler.

Technical Details
------
My thinking with this PR is that using `CRM_Utils_Array::value()` in a unit test smells sloppy. Unit tests usually want to strictly assert values and don't need/shouldn't use the hedge provided by that function. So I've removed it more aggressively in the test classes than I would from runtime code (without providing the extra `??` hedge unless the test actually needs it).

colemanw committed:
Cleanup array syntax in tests directory

eileenmcnaughton commented:
@colemanw 4 fails - give  the degree to which this is likely to go stale / make others stale maybe just unchange those 2 classes & we can get the rest merged?

[Test Result](https://test.civicrm.org/job/CiviCRM-Core-Matrix-PR/5648/BKPROF=dfl,SUITES=phpunit-crm,label=bknix-tmp/testReport/) (4 failures / +4)
[CRM_Core_BAO_FinancialTrxnTest.testUpdateCreditCardDetailsUsingContributionAPI](https://test.civicrm.org/job/CiviCRM-Core-Matrix-PR/5648/BKPROF=dfl,SUITES=phpunit-crm,label=bknix-tmp/testReport/junit/(root)/CRM_Core_BAO_FinancialTrxnTest/testUpdateCreditCardDetailsUsingContributionAPI/)
[CRM_Core_BAO_FinancialTrxnTest.testUpdateCreditCardDetails](https://test.civicrm.org/job/CiviCRM-Core-Matrix-PR/5648/BKPROF=dfl,SUITES=phpunit-crm,label=bknix-tmp/testReport/junit/(root)/CRM_Core_BAO_FinancialTrxnTest/testUpdateCreditCardDetails/)
[CRM_Utils_StringTest.testExtractName](https://test.civicrm.org/job/CiviCRM-Core-Matrix-PR/5648/BKPROF=dfl,SUITES=phpunit-crm,label=bknix-tmp/testReport/junit/(root)/CRM_Utils_StringTest/testExtractName/)
[CRM_Utils_SystemTest.testRedirectHook with data set #2](https://test.civicrm.org/job/CiviCRM-Core-Matrix-PR/5648/BKPROF=dfl,SUITES=phpunit-crm,label=bknix-tmp/testReport/junit/(root)/CRM_Utils_SystemTest/testRedirectHook_with_data_set__2/)

colemanw commented:
@eileenmcnaughton I've updated the tests. My thinking with this PR is that using `CRM_Utils_Array::value()` in a unit test smells sloppy. Unit tests usually want to strictly assert values and don't need/shouldn't use the hedge provided by that function. So I removed it more aggressively in the test classes than I would from runtime code just to see what would happen. 4 failures isn't too bad!
